docs/advanced-features/custom-directories.md

```diff
@@ -0,0 +1,42 @@
+---
+description: Customize the directories included in your Next.js project.
+---
+
+# Custom Directories
+
+If you need to customize which directories are included in your project, i.e. to
+prevent redundant type-checking during `next build`, you can adjust the
+`include` and `exclude` fields of your root-level tsconfig.json to do so:
+
+```none
+{
+  "exclude": ["excludeDir/"],
+  ...
+}
+```
+
+Alternatively, you can override the `include` field for the root-level tsconfig
```

Comment:
It would be better with an example of what should absolutely be present in a Next.js project, like "pages". Btw, I am not sure what's the absolute minimum set of included folder is needed in a Next.js project. Probably just pages? Since pages will import all relevant components

Reply:
This was the main reason I also recommended to the team that they consider embracing `src/` structure as default, because we could literally just do `"rootDir": "src/"` and this entire question answers itself.

But yes I think the appropriate default would be `"include": ["src/", "pages/"]` to handle either root-level `pages/` or `src/`-dir config.